crates/signer-ledger/src/signer.rs

```diff
@@ -27,10 +28,37 @@ pub struct LedgerSigner {
     pub(crate) address: Address,
 }
 
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+impl alloy_network::TxSigner<Signature> for LedgerSigner {
+    #[inline]
+    async fn sign_transaction(
+        &self,
+        tx: &mut dyn SignableTransaction<Signature>,
+    ) -> Result<Signature> {
+        let chain_id = match (self.chain_id(), tx.chain_id()) {
```

Comment:
agree w dani comment on macro to reduce code dup, or function

Reply:
done with a macro